ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                                          hMMFeature.pRecords[0].nNumField))
+                return OGRERR_NOT_ENOUGH_MEMORY;
+
+            hMMFeature.pRecords[0].pField[iField].dValue =
+                poFeature->GetFieldAsDouble(iField);
+            if (MM_SecureCopyStringFieldValue(
+                    &hMMFeature.pRecords[0].pField[iField].pDinValue,
+                    poFeature->GetFieldAsString(iField),
+                    &hMMFeature.pRecords[0].pField[iField].nNumDinValue))
+                return OGRERR_NOT_ENOUGH_MEMORY;
+            hMMFeature.pRecords[0].pField[iField].bIsValid = 1;
+        }
+        else
+        {
+            CPLError(CE_Failure, CPLE_NotSupported,
+                     "MiraMon: Field type %d not processed by MiraMon\n",
+                     eFType);
+            return OGRERR_UNSUPPORTED_GEOMETRY_TYPE;
+        }
+    }
+
+    return OGRERR_NONE;
+}
+
+/****************************************************************************/
+/*                             GetLayerDefn()                               */
+/*                                                                          */
+/****************************************************************************/
+OGRFeatureDefn *OGRMiraMonLayer::GetLayerDefn()
+{
+    return poFeatureDefn;
+}
+
+/****************************************************************************/
+/*                             GetExtent()                                  */
+/*                                                                          */
+/*      Fetch extent of the data currently stored in the dataset.           */
+/*      The bForce flag has no effect on SHO files since that value         */
+/*      is always in the header.                                            */
+/****************************************************************************/
+
+OGRErr OGRMiraMonLayer::GetExtent(OGREnvelope *psExtent, int bForce)
+
+{
+    if (phMiraMonLayer)
+    {
+        psExtent->MinX = phMiraMonLayer->TopHeader.hBB.dfMinX;
+        psExtent->MaxX = phMiraMonLayer->TopHeader.hBB.dfMaxX;
+        psExtent->MinY = phMiraMonLayer->TopHeader.hBB.dfMinY;
+        psExtent->MaxY = phMiraMonLayer->TopHeader.hBB.dfMaxY;
+    }
+    else
+    {
+        if (!bForce)
+            return OGRERR_FAILURE;
+    }
+
+    return OGRERR_NONE;
+}
+
+/****************************************************************************/
+/*                           TestCapability()                               */
+/****************************************************************************/
+
+int OGRMiraMonLayer::TestCapability(const char *pszCap)
+
+{
+    if (EQUAL(pszCap, OLCRandomRead))
+        return TRUE;
+
+    if (EQUAL(pszCap, OLCSequentialWrite))
+        return TRUE;
+
+    if (EQUAL(pszCap, OLCFastFeatureCount))
+        return TRUE;
+
+    if (EQUAL(pszCap, OLCFastGetExtent))
+        return TRUE;
+
+    if (EQUAL(pszCap, OLCCreateField))
+        return TRUE;
```

Comment:
is it possible to create fields after features have been created ? if not, it should return false if there are features
```suggestion
        return m_bUpdate;
```

Reply:
Done